### PR TITLE
fix: add unitest

### LINKED ISSUE #1914: pkg/util unitest
Full coverage of pkg/utils

### PR DESCRIPTION
**Pull request by lut777**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Bug fixes

#### Which issue(s) this PR fixes:
Fixes #1914 

image
image
image

files which could be added with unit tests are added. Rest of them need some dummy k8s/docker dummy client.

### CODE REVIEW

File: pkg/util/address_family_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		{
+			name: "err",
+			prot: "damn",
+			want: unix.AF_UNSPEC,
+			err:  errors.New(""),
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			ans, err := ProtocolToFamily(c.prot)
```
Comment (oilbeater) [inline pkg/util/address_family_test.go:44]: if ans != c.want || err != c.err {}

File: pkg/util/address_family_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			if c.err == nil && err != nil {
+				t.Fatalf("%v expected %v, but %v, %v got",
```
Comment (oilbeater) [inline pkg/util/address_family_test.go:46]: why use `t.Fatalf` here and `t.Errorf` in other places?

File: pkg/util/net_test.go [hunk 3]
```diff
@@ -92,3 +101,925 @@ func TestCheckCIDRSpec(t *testing.T) {
...
+	}{
+		{
+			name: "correct",
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			ans := GenerateMac()
+			if _, err := net.ParseMAC(ans); err != nil {
+				t.Fatalf("%v expected %v, but %v got",
```
Comment (oilbeater) [inline pkg/util/net_test.go:118]: Should reconsider all the `t.Fatalf`

File: pkg/util/net_test.go [hunk 4]
```diff
@@ -92,3 +101,925 @@ func TestCheckCIDRSpec(t *testing.T) {
...
+			name:   "controversy",
+			subnet: "192.168.0.23/32",
+			expect: "192.168.0.24",
+			err:    nil,
+		},
+		{
+			name:   "subneterr",
+			subnet: "192.168.0.0",
+			expect: "192.168.0.1",
+			err:    errors.New(""),
```
Comment (oilbeater) [inline pkg/util/net_test.go:251]: error value should also be tested

File: pkg/util/k8s_test.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+func TestGetNodeInternalIP(t *testing.T) {
+	tests := []struct {
+		name string
+		node v1.Node
+		exp4 string
+		exp6 string
+	}{
+		{
+			name: "correct",
```
Comment (oilbeater) [inline pkg/util/k8s_test.go:17]: case name should be meaningful

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/util/address_family_test.go (hunk 2 above)
Comment (lut777) [reply pkg/util/address_family_test.go:46]: fixed...copied from previous work and didn't notice that diff.

File: pkg/util/net_test.go (hunk 3 above)
Comment (lut777) [reply pkg/util/net_test.go:118]: fixed

File: pkg/util/address_family_test.go (hunk 1 above)
Comment (lut777) [reply pkg/util/address_family_test.go:44]: fixed with a new function to identify the error

File: pkg/util/net_test.go (hunk 4 above)
Comment (lut777) [reply pkg/util/net_test.go]: now all error message is included in the unitest

File: pkg/util/k8s_test.go (hunk 5 above)
Comment (lut777) [reply pkg/util/k8s_test.go:17]: fixed.

File: pkg/util/address_family_test.go [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+			prot: "damn",
+			want: -1,
+			err:  "invalid protocol",
+		},
+	}
+	for _, c := range tests {
+		t.Run(c.name, func(t *testing.T) {
+			ans, err := ProtocolToFamily(c.prot)
+			if ans != c.want || !ErrorContains(err, c.err) {
+				t.Fatalf("%v expected %v, %v, but %v, %v got",
```
Comment (oilbeater) [inline pkg/util/address_family_test.go:46]: use `Errorf` to continue test

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/util/address_family_test.go (hunk 6 above)
Comment (lut777) [reply pkg/util/address_family_test.go]: done

**Review by oilbeater — APPROVED** [review]

**Pull request merged**